File: Reverse-Integer.py
1class Solution(object):
2    def reverse(self, x):
3        """
4        :type x: int
5        :rtype: int
6        """
7        sign = -1 if x < 0 else 1
8        x = abs(x)
9        
10        rev = 0
11        while x != 0:
12            digit = x % 10
13            rev = rev * 10 + digit
14            x //= 10
15        
16        rev *= sign
17        
18        if rev < -2**31 or rev > 2**31 - 1:
19            return 0
20        
21        return rev
22        

File: Reverse-Integer.cpp
1class Solution {
2public:
3    int reverse(int x) {
4        int rev = 0;
5        
6        while (x != 0) {
7            int digit = x % 10;
8            x /= 10;
9
10            if (rev > INT_MAX / 10 || rev < INT_MIN / 10) {
11                return 0;
12            }
13
14            rev = rev * 10 + digit;
15        }
16        
17        return rev;
18    }
19};